### ISSUE TITLE
How are models synchronized across GPUS

### ISSUE THREAD

**Issue by ranran9991** [KaihuaTang/Scene-Graph-Benchmark.pytorch#51]
## ❓ Questions and Help
How are the models and gradients synchronized when training in the multigpu regime?
I get that each process has its own copy of the model and that each process gets its own batch every iteration. When and where are the models synchronized across the gpus? do they have the same starting weights?

Would love a clarification on how the distributed training works in this repo as I didn't fully understand just by looking at the code

**Comment by ranran9991** [comment]
Still struggling to figure this out

**Comment by KaihuaTang** [comment]
The code is built on top of https://github.com/facebookresearch/maskrcnn-benchmark. Maybe you can approach their instructions for the multi-GPU training questions.
